### PR TITLE
minor: Sync from rust

### PR DESCRIPTION
**Pull request by lnicola**

### CODE REVIEW

File: crates/syntax/src/tests.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ fn benchmark_parser() {
         let _b = bench("parsing");
         let p = SourceFile::parse(&data);
         assert!(p.errors.is_empty());
-        assert_eq!(p.tree().syntax.text_range().len(), 352474.into());
+        assert_eq!(p.tree().syntax.text_range().len(), 352458.into());
```
Comment (lnicola) [inline crates/syntax/src/tests.rs:42]: Because of the `fmt` changes, maybe?

**Comment by lnicola** [comment]
Dropped the two commits from https://github.com/rust-lang/rust/pull/109002 because they break building against the stable sysroot.

**Comment by lnicola** [comment]
Actually, according to https://github.com/rust-lang/rust-analyzer/pull/13676#issuecomment-1327662084, it seems we can't drop commits that way.

**Comment by Veykril** [comment]
Hmm, can we add those commits and merge them in without CI approval? Then fix them up afterwards?

**Comment by Veykril** [comment]
Ahh nevermind I see the issue now

**Comment by lnicola** [comment]
Yes, but we'll have to propagate our "fix" downstream, if it involves changing the code there :smile: .

**Pull request closed without merging**